ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -0,0 +1,3018 @@
+          completionBlock([NSUnderlyingErrorKey: error])
+        }
+        NotificationCenter.default.post(name: .keymanLanguagesDownloadFailed, object: self,
+                                        userInfo: [NSUnderlyingErrorKey: error])
+      }
+    }
+  }
+
+  func downloadFailed(forKeyboard keyboardInfo: [AnyHashable: Any], error: NSError) {
+    let kbInfo = keyboardInfo[Key.keyboardInfo] ?? keyboardInfo
+    NotificationCenter.default.post(name: .keymanKeyboardDownloadFailed, object: self,
+                                    userInfo: [
+                                      Key.keyboardInfo: kbInfo,
+                                      NSUnderlyingErrorKey: error
+      ])
+  }
+
+  // MARK: - Loading custom keyboards
+
+  private func preloadFile(srcUrl: URL, dstDir dirUrl: URL, shouldOverwrite: Bool) {
+    let dstUrl = dirUrl.appendingPathComponent(srcUrl.lastPathComponent)
+    do {
+      if !FileManager.default.fileExists(atPath: dstUrl.path) {
+        try FileManager.default.copyItem(at: srcUrl, to: dstUrl)
+      } else if shouldOverwrite {
+        try FileManager.default.removeItem(at: dstUrl)
+        try FileManager.default.copyItem(at: srcUrl, to: dstUrl)
+      } else {
+        kmLog("File already exists at \(dstUrl) and not overwriting", checkDebugPrinting: false)
+        return
+      }
+      addSkipBackupAttribute(to: dstUrl)
+    } catch {
+      kmLog("Error copying file: \(error)", checkDebugPrinting: false)
+    }
+  }
+
+  /// Preloads a .js file for a language so that the keyboard is available without downloading.
+  /// - Precondition:
+  ///   - The .js filename must remain the same as when obtained from Keyman.
+  ///   - The .js file must be bundled in your application.
+  public func preloadLanguageFile(atPath languagePath: String, shouldOverwrite: Bool) {
+    guard let languageDir = activeLanguageDirectory() else {
+      kmLog("Could not find/create the Keyman language directory", checkDebugPrinting: false)
+      return
+    }
+    preloadFile(srcUrl: URL.init(fileURLWithPath: languagePath),
+                dstDir: languageDir,
+                shouldOverwrite: shouldOverwrite)
+  }
+
+  /// Preloads a .ttf or .otf file to be available without downloading.
+  /// - Precondition:
+  ///   - The font file must be bundled in your application.
+  /// - SeeAlso: `registerCustomFonts()`
+  public func preloadFontFile(atPath fontPath: String, shouldOverwrite: Bool) {
+    guard let fontDir = activeFontDirectory() else {
+      kmLog("Could not find/create the Keyman font directory", checkDebugPrinting: false)
+      return
+    }
+    preloadFile(srcUrl: URL.init(fileURLWithPath: fontPath),
+                dstDir: fontDir,
+                shouldOverwrite: shouldOverwrite)
+  }
+
+  /// Registers all new fonts found in the font path. Call this after you have preloaded all your font files
+  /// with `preloadFontFile(atPath:shouldOverwrite:)`
+  public func registerCustomFonts() {
+    // T
```

Comment:
Tested on iPad and noticed that the views resigned first responder appropriately. The reason this is removed is because `UIApplication.shared` can't be used in an extension and that `findFirstResponder` is a private API.

Reply:
Correction: `findResponder` is the private API, but we don't use it. `findFirstResponder` is an extension in `UIView+Helpers.m`. `UIApplication.shared` is still best to be avoided though.